CRM/Case/Form/Activity.php

```diff
@@ -203,10 +203,11 @@ public function setDefaultValues() {
       }
       $this->assign('targetContactValues', empty($targetContactValues) ? FALSE : $targetContactValues);
 
-      if (isset($this->_encounterMedium)) {
-        $this->_defaults['medium_id'] = $this->_encounterMedium;
+      if ($this->_activityId) {
+        $this->_defaults['medium_id'] = CRM_Core_DAO::getFieldValue('CRM_Activity_DAO_Activity', $this->_activityId, 'medium_id');
       }
```

Comment:
I don't even know why this shenanigan still exists. Why wouldn't this value be set along with all the others by the call to `parent::setDefaultValues()` above?

Reply:
There is no medium on regular activities.

Reply:
The field exists though so shouldn't it be populated?
https://github.com/civicrm/civicrm-core/blob/baffc20664713892270c668eac242dc91b95c66f/xml/schema/Activity/Activity.xml#L250-L264

Reply:
I'm just saying that's why it's not unexpected to see encounter_medium code here.

Reply:
I tested on a site and it does seem to work without those lines of code (`$this->_defaults['medium_id']   = ..`)

Reply:
I checked the code and it certainly looks like the lines aren't needed. The `setDefaultValues` function does a pretty standard `DAO->find(TRUE)` and `CRM_Core_DAO::storeValues()` which is basically like `SELECT *`.

So it *should* work without it, and it *does* work without it. Let's remove it :)

Reply:
Yep, I had a side-quest (joys of numeric custom fields), and now updated this PR to remove it.